File: tests/python/tests-python-selene/mock_scenarios.py
"""WireMock scenario switch — mirrors java `api.MockScenarios`."""

from __future__ import annotations

import httpx

from config import TestConfig

_TIMEOUT_S = 5


def _admin(config: TestConfig, path: str) -> str:
    return config.api_base_url.rstrip("/") + path


def available(config: TestConfig) -> bool:
    """True when the stand exposes WireMock admin (mock profile only)."""
    try:
        response = httpx.get(_admin(config, "/__admin/scenarios"), timeout=_TIMEOUT_S)
        return response.status_code == 200
    except httpx.HTTPError:
        return False


def set_state(config: TestConfig, scenario: str, state: str) -> None:
    response = httpx.put(
        _admin(config, f"/__admin/scenarios/{scenario}/state"),
        json={"state": state},
        timeout=_TIMEOUT_S,
    )
    assert response.status_code == 200, response.text


def reset_all(config: TestConfig) -> None:
    response = httpx.post(_admin(config, "/__admin/scenarios/reset"), timeout=_TIMEOUT_S)
    assert response.status_code == 200, response.text

File: tests/csharp/tests-csharp-nunit-restsharp-selenium/Api/MockScenarios.cs
using Allure.NUnit.Attributes;
using Config;
using RestSharp;

namespace Api;

/// <summary>
/// WireMock scenario switch for the mock stand (compose profile <c>mock</c>; the gateway
/// proxies <c>/__admin/</c> to WireMock).
/// </summary>
public static class MockScenarios
{
    public static bool Available()
    {
        try
        {
            var response = RestSharpHttp.Request(Method.Get, "/__admin/scenarios");
            return response.Status == 200;
        }
        catch (Exception)
        {
            return false;
        }
    }

    [AllureStep("Mock: switch scenario {scenario} to state {state}")]
    public static void SetState(string scenario, string state)
    {
        var origin = ConfigReader.ResolveApiBaseUrl().TrimEnd('/');
        using var client = new RestClient(origin);
        var request = new RestRequest($"/__admin/scenarios/{scenario}/state", Method.Put);
        request.AddJsonBody(new { state });
        var response = client.Execute(request);
        Assert.That((int)response.StatusCode, Is.EqualTo(200), response.Content);
    }

    [AllureStep("Mock: reset all scenarios")]
    public static void ResetAll()
    {
        var origin = ConfigReader.ResolveApiBaseUrl().TrimEnd('/');
        using var client = new RestClient(origin);
        var request = new RestRequest("/__admin/scenarios/reset", Method.Post);
        var response = client.Execute(request);
        Assert.That((int)response.StatusCode, Is.EqualTo(200), response.Content);
    }
}
